internal/cli/compile/compile.go

```diff
@@ -363,12 +362,8 @@ func runCompileCommand(cmd *cobra.Command, args []string) {
 				panic(tr("Platform ID is not correct"))
 			}
 
-			// FIXME: Here we should not access PackageManager...
 			pme, release := commands.GetPackageManagerExplorer(compileRequest)
-			platform := pme.FindPlatform(&packagemanager.PlatformReference{
-				Package:              split[0],
-				PlatformArchitecture: split[1],
-			})
```

Comment:
The problem is not the use of `packagemanager.PlatformReference` but the use of `packagemanager.Explorer`. The call to `pme.FindPlatform` must be replaced with the equivalent gRPC call. Basically, the final goal is to remove the `commands.GetPackageManagerExplorer`, that allows bypassing the gRPC API.

Once the call to `pme.FindPlatform` is removed the change to `FindPlatform` can be reverted.